blocks/library/embed/index.js

```diff
@@ -26,7 +27,33 @@ import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 // These embeds do not work in sandboxes
 const HOSTS_NO_PREVIEWS = [ 'facebook.com' ];
 
-function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, keywords = [] } ) {
+// Caches the embed API calls, so if blocks get transformed, or deleted and added again, we don't spam the API.
+const wpEmbedAPI = memoize( ( url ) => wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } ) );
+
+const matchesPatterns = ( url, patterns ) => {
```

Comment:
This pattern of logic is what [`Array#some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is perfectly suited for:

```js
return patterns.some( ( pattern ) => (
	url.match( new RegExp( '^https?:\/\/' + patterns[ patternIndex ] + '$', 'i' ) )
) );
```